docs/api-reference.md

```diff
@@ -0,0 +1,232 @@
+
+- **Lifting a Pure Value**
+
+  ```scala
+  import cats.Id
+  import pl.iterators.sealedmonad._
+  import pl.iterators.sealedmonad.syntax._
+
+  // Lifts an integer into the Sealed context.
+  val sealedValue: Sealed[Id, Int, Nothing] = 42.liftSealed[Id, Nothing]
+  println(sealedValue.run) // prints 42
+  ```
+
+- **Lifting an Effectful Computation**
+
+  ```scala
+  import cats.effect.IO
+  import pl.iterators.sealedmonad._
+  import pl.iterators.sealedmonad.syntax._
+
+  // Creates a Sealed from an IO computation.
+  val ioSealed: Sealed[IO, String, Nothing] = IO("Hello, World!").seal
+  ioSealed.run.map(println) // prints "Hello, World!"
+  ```
+
+---
+
+## Core Operations
+
+### Mapping (map)
+
+Transforms the intermediate value using a pure function. If the computation is successful, the function is applied; otherwise the final value is preserved.
+
+```scala
+import cats.Id
+import pl.iterators.sealedmonad._
+import pl.iterators.sealedmonad.syntax._
+
+val sealedInt: Sealed[Id, Int, String] = 10.liftSealed[Id, String]
+val sealedMapped: Sealed[Id, Int, String] = sealedInt.map(_ + 5)
+// Since the value is present, run returns the new intermediate value.
+println(sealedMapped.run) // prints 15
+```
+
+### FlatMapping (flatMap)
+
+Chains computations where the next step may also return a Sealed instance. This operation maintains the short-circuiting behavior in case any step produces a final ADT value.
+
+```scala
+import cats.Id
+import pl.iterators.sealedmonad._
+import pl.iterators.sealedmonad.syntax._
+
+def safeDivide(a: Int, b: Int): Sealed[Id, Int, String] =
+  if (b != 0) (a / b).liftSealed[Id, String]
+  else "DivisionByZero".seal
+
+val result = 20.liftSealed[Id, String].flatMap(x => safeDivide(x, 4))
+println(result.run) // prints 5
+
+val errorResult = 20.liftSealed[Id, String].flatMap(x => safeDivide(x, 0))
+println(errorResult.run) // prints DivisionByZero
+```
+
+### Effectful Transformation (semiflatMap)
+
+Transforms the intermediate value using an effectful function; that is, a function returning a value in an effect context F.
+
+```scala
+import cats.effect.IO
+import pl.iterators.sealedmonad._
+import pl.iterators.sealedmonad.syntax._
+
+val ioSealed: Sealed[IO, Int, String] = IO(50).seal
+val transformed: Sealed[IO, Int, String] = ioSealed.semiflatMap(i => IO(i * 2))
+transformed.run.map(println) // prints 100
+```
+
+### Converting Options to Sealed (valueOr / valueOrF)
+
+Transforms an effectful Option into a Sealed instance. If the Option is defined, its value becomes the intermediate result; otherwise, the supplied ADT (or effectful ADT) is used to short-circuit.
+
+```scala
+import cats.effect.IO
+import pl.iterators.sealedmonad._
+import pl.iterators.sealedmonad.syntax._
+
+val someUser: IO[Option[String]] = IO.pure(Some("JohnDoe"))
+val missingUser: IO[Option[String]] = IO.pure(None)
+
+// Use a plain ADT to signal error
+val sealedUser1: Sealed[IO, String, String] = someUser.valueOr("UserNotFound")
```

Comment:
I think it's pretty confusing that it uses String for intermediate and String as ADT